infra_event_notifier/cli/util.py

```diff
@@ -0,0 +1,31 @@
+import os
+from pathlib import Path
+from typing import cast
+
+_workspace_root = None
+
+
+def workspace_root(cwd=Path.cwd()) -> Path:
+    """
+    Finds the root directory of the workspace and caches it.
+    """
+    global _workspace_root
```

Comment:
Please let's not replicate the bad practices from sentry-kube. We are having a hard time eradicating these global variables.

If you need to pass the workspace around, please pass it as a parameter where you need it rather than initializing a global variable. 

Most importantly. I don't see a need for this script to detect the root by itself. Just require `SENTRY_KUBE_ROOT` to be populated

Reply:
Yes, I think that's reasonable. I can make that change in the ops devenv.

Reply:
getsentry/ops/pull#12153, let me update this logic

Reply:
Tested and confirmed working now